src/rules/extensions.js

```diff
@@ -37,15 +39,15 @@ module.exports = function (context) {
     const extension = path.extname(resolvedPath || importPath).substring(1)
 
     if (!extension || !endsWith(importPath, extension)) {
-      if (isUseOfExtensionEnforced(extension)) {
+      if (isUseOfExtensionRequired(extension) && !isUseOfExtensionForbidden(extension)) {
```

Comment:
You could compute the value for the extension `modifiers[extension] || defaultConfig` once, instead of recomputing it several times (nitpick)